### PR TITLE
Initial changes for reusable QDS report

### PR DESCRIPTION
**Pull request by kisantia**
This PR has the initial changes for building a reusable report in the QDS extension using Modelview components. The two placeholder reports being added in this PR are Overall Resource Consumption and Top Resource Consuming Queries. Right now it has placeholder text components in flex containers to show the different sections in the reports, but those will be replaced with chart/table/query plan components in the future. 
image
image

The entry points for these reports are currently commands in the command palette. Updating the reports to be opened with the context of a database is a task for later

The `BaseQueryStoreReport` class supports resizable components by using splitviews, and also supports building non-resizeable reports (for example, the Overall Resource Consumption report is not resizable in SSMS).

initialReports

### CODE REVIEW

File: extensions/query-store/src/extension.ts [hunk 1]
```diff
@@ -4,8 +4,13 @@
  *--------------------------------------------------------------------------------------------*/
 
 import * as vscode from 'vscode';
+import { TopResourceConsumingQueries } from './reports/topResourceConsumingQueries';
+import { OverallResourceConsumption } from './reports/overallResourceConsumption';
 
-export async function activate(_context: vscode.ExtensionContext): Promise<void> {
+export async function activate(context: vscode.ExtensionContext): Promise<void> {
+	// TODO: get db name
+	context.subscriptions.push(vscode.commands.registerCommand('queryStore.topResourceConsumingQueriesOpen', async () => { await new TopResourceConsumingQueries(context, 'WideWorldImporters').open() }));
```
Comment (lewis-sanchez) [inline extensions/query-store/src/extension.ts:12]: Nit: Declare a const with the hardcoded database name below the TODO.

File: extensions/query-store/src/reports/baseQueryStoreReport.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+	 * @param containers Array of containers to add to the main container
+	 * @returns FlexContainer or SplitViewContainer containing the containers
+	 */
+	private async createMainContainer(view: azdata.ModelView, containers: azdata.FlexContainer[]): Promise<azdata.FlexContainer | azdata.SplitViewContainer> {
+		let mainContainer;
+
+		switch (containers.length) {
+			case 1: {
+				mainContainer = view;
+			}
```
Comment (lewis-sanchez) [inline extensions/query-store/src/reports/baseQueryStoreReport.ts:59]: Did you mean to omit the `break`? It's currently going to fall through to the next case without it.

**Review by aasimkhan30 — APPROVED** [review]
Except for the break in switch case, rest of the code looks good for initial check in .

**Review by lewis-sanchez — APPROVED** [review]
Looks good except for the break.

File: extensions/query-store/src/reports/baseQueryStoreReport.ts (hunk 2 above)
Comment (kisantia) [reply extensions/query-store/src/reports/baseQueryStoreReport.ts:60]: whoops good catch! I added this case later after the others and missed it

File: extensions/query-store/src/extension.ts (hunk 1 above)
Comment (kisantia) [reply extensions/query-store/src/extension.ts:12]: the database name/connection profile will later get passed into the command from where the command is launched from, so it'll look like `context.subscriptions.push(vscode.commands.registerCommand('queryStore.topResourceConsumingQueriesOpen', async (profile: azdata.IConnectionProfile) => { await new TopResourceConsumingQueries(context, profile).open() }));`

**Pull request merged**